startstoptorq.sh

```diff
@@ -0,0 +1,275 @@
+  if [[ `echo $proc | grep "unavailable"` ]]; then
+    echo $proc
+  else 
+    sline=$(startline $1)                                                                             # line to run each process in debug mode
+    eval "$sline -debug"                                                                              # append flag to start in debug mode
+  fi
+ }
+
+summary() {
+  if [ -z `findproc $1` ]; then                                                                     # check process not running
+    printf "%-8s | %-14s | %-6s |\n" `date '+%H:%M:%S'` "$1" "down"                                 
+  else
+    pid=$(findproc $1)
+    port=`netstat -pl 2>/dev/null | grep $pid | awk '{ print $4 }' | head -1 | cut -c 3-`
+    printf "%-8s | %-14s | %-6s | %-6s | %-6s\n" `date '+%H:%M:%S'` "$1" "up" "$port" "$pid"                      
+  fi
+ }
+
+stop() {
+  if [ -z `findproc $1` ]; then                                                                     # check process not running
+    echo `date '+%H:%M:%S'` "| $1 is not currently running"
+  else
+    echo `date '+%H:%M:%S'` "| Shutting down $1..."
+    pid=$(findproc $1)
+    eval "kill -15 $pid"                                                                            # kill process pid
+  fi
+ }
+
+getall() {
+  procs=`awk -F, '{if(NR>1) print $4}' $csvpath`                                                    # get all processes from csv
+  start=""
+  for a in $procs;
+  do
+    procno=`awk '/'$a'/{print NR}' $csvpath`  		                                            # get line number for file
+    f=`getfield $procno startwithall`
+    if [ "1" == "$f" ]; then
+      start="$start $a"
+    fi
+  done
+  echo $start
+ }
+
+checkinput() {
+  input=$*                                                                                          # get all input process names
+  procs=$(getall)                                                                                   # get all process names from csv
+  avail=()
+  for i in $input;
+  do 
+    if [[ `echo "$procs" | grep -w "$i"` ]]; then                                                   # check input process is valid
+      avail+="$i "                                                                                  # get only valid processes
+    else 
+      echo `date '+%H:%M:%S'` "| $i failed - unavailable processname"
+    fi
+  done
+  procs=$avail                                                                                      # assign valid processes
+ }
+
+getprocs() {
+  if [ "$2" == "all" ]; then 
+    procs=$(getall)                                                                                 # get all processes
+  else
+    shift                                                                                           # ignore first argument
+    checkinput $@                                                                                   # check input process names
+  fi
+ }
+
+flag() {
```

Comment:
this function could maybe do with an explanation, not very clear what's going on here